src/odemis/gui/comp/viewport.py

```diff
@@ -1065,7 +1066,334 @@ def _on_peak_method(self, state):
+                              self.canvas.pos_y_to_val_y(pos[1]))
+
+        logging.debug("Drag started at %s", self.drag_init_pos)
+
+    def on_drag_end(self, evt):
+        """ End the dragging procedure """
+        if not self.canvas.has_data():
+            return
+
+        # If the canvas was being dragged
+        if self._dragging:
+            self._dragging = False
+
+        self.was_dragged = False
+
+    def on_hlegend_motion(self, evt):
+        """ Process mouse motion
+
+        Set the drag shift and refresh the image if dragging is enabled and the left mouse button is
+        down.
+
+        """
+        if self._dragging:
+
+            v_pos = (self.canvas.pos_x_to_val_x(evt.Position[0]),
+                self.canvas.pos_y_to_val_y(evt.Position[1]))
+
+            self.drag_shift = (v_pos[0] - self.drag_init_pos[0],
+                          v_pos[1] - self.drag_init_pos[1])
+
+            # Rescale the horizontal axis
+            (lo, hi) = self.hrange.value
+
+            lo += (self.drag_shift[0] * self._hdrag_scale_factor)
+            hi += (self.drag_shift[0] * self._hdrag_scale_factor)
+
+            if lo < self.canvas.data_xrange[0]:
+                lo = self.canvas.data_xrange[0]
+                hi = self.canvas.display_xrange[1]
+            if hi > self.canvas.data_xrange[1]:
+                hi = self.canvas.data_xrange[1]
+                lo = self.canvas.display_xrange[0]
+
+            self.hrange.value = (lo, hi)
+
+    def on_vlegend_motion(self, evt):
+        """ Process mouse motion
+
+        Set the drag shift and refresh the image if dragging is enabled and the left mouse button is
+        down.
+
+        """
+        if self._dragging:
+
+            v_pos = (self.canvas.pos_x_to_val_x(evt.Position[0]),
+                self.canvas.pos_y_to_val_y(evt.Position[1]))
+
+            self.drag_shift = (v_pos[0] - self.drag_init_pos[0],
+                          v_pos[1] - self.drag_init_pos[1])
+
+            # Rescale the vertical axis
+            (lo, hi) = self.vrange.value
+            lo += (self.drag_shift[1] * self._vdrag_scale_factor)
+            hi += (self.drag_shift[1] * self._vdrag_scale_factor)
+
+            if lo < self.canvas.data_yrange[0]:
+                lo = self.canvas.data_yrange[0]
+                hi = self.canvas.display_yrange[1]
+            if hi > self.canvas.data_yrange[1] + self.canvas.data_yrange[1] * self._vmargin:
+                hi = self.canvas.data_yrange[1] + self.canvas.data_yrange[1] * self._vmargin
+                lo = self.canvas.display_yrange[0]
+
+            self.vrange.value = (lo, hi)
+
+    def on_canvas_motion(self, evt):
+        """ Process mouse motion
+
+        Set the drag shift and refresh the image if dragging is enabled and the left mouse button is
+        down.
+
+        """
+        if self._dragging:
+
+            v_pos = (self.canvas.pos_x_to_val_x(evt.Position[0]),
```

Comment:
Isn't it possible to use `v_pos = self.canvas.pos_to_val(evt.Position)`?

Reply:
Apparently it doesn't have it - which is strange. I left it the way it was.